Demo e2e annotations

andrii-harbour opened the pull request.

andrii-harbour committed:
feat: enhance demo with signature functionality and API integration

- Added a custom signature component using `signature_pad` for drawing signatures.
- Updated the demo to include document field definitions and improved event logging.
- Integrated API calls for document annotation and signing, utilizing the provided API key.
- Enhanced the Vite configuration for better development experience and added a README for setup instructions.
- Included new dependencies in `package.json` and updated the lock file accordingly.

andrii-harbour committed:
fix: fixed payload

caio-pizzol commented on demo/vite.config.ts:
```diff
@@ -1,10 +1,25 @@
+    }),
   },
-});
+  server: {
```
is this really needed? where's the backend?

caio-pizzol commented on demo/src/App.tsx:
```diff
@@ -1,42 +1,424 @@
+    const headers: Record<string, string> = {
+      'Content-Type': 'application/json',
+    };
+    const bearerToken = import.meta.env.VITE_SUPERDOC_SERVICES_API_KEY;
```
The `VITE_` prefix bakes this into the JS bundle—anyone can extract it from the deployed site. Consider a Cloudflare Worker or similar proxy to keep the key server-side.

caio-pizzol commented on demo/src/App.tsx:
```diff
@@ -1,42 +1,424 @@
+      const { base64: signedBase64 } = signResult.document;
+
+      // Step 4: Download the signed PDF
+      const byteCharacters = atob(signedBase64);
```
The base64→blob conversion is duplicated. Consider extracting to a utility function like `base64ToBlob(base64, contentType)`.

caio-pizzol reviewed (commented):
Looking good!

Main concerns are API key security and demo project structure

andrii-harbour committed:
feat: added proxy-server for the demo and basic workflow to test deployment

- Changed the proxy target in Vite config to point to localhost for local testing.
- Removed the custom signature component and replaced it with a simpler implementation.
- Introduced a helper function to download response blobs as files.
- Refactored API calls in the App component to streamline document signing and downloading processes.

andrii-harbour committed:
test: temporary changing existing workflow for testing

andrii-harbour committed:
fix: used wrong workflow

andrii-harbour committed:
fix: replaced key var with secret

andrii-harbour committed:
fix: added cors, fixed ci/cd process

andrii-harbour committed:
fix: fixed linter config for server

andrii-harbour committed:
fix: fixed workflows for github

andrii-harbour committed:
fix: fixed linter for node server

andrii-harbour committed:
chore: cleaning

andrii-harbour committed:
chore: server using pnpm instead of npm

caio-pizzol reviewed (commented):
Looking good!!

Just some small tweaks

andrii-harbour committed:
docs: update README with demo integration details and setup instructions

andrii-harbour committed:
feat: add demo user and IP address handling in server and update signer information

caio-pizzol reviewed (approved):
LGTM
